test/LibraryManager.Test/LibraryNamingSchemeTest.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.Web.LibraryManager.Helpers;
+
+namespace Microsoft.Web.LibraryManager.Test
+{
+    [TestClass]
+    public class LibraryNamingSchemeTest
+    {
+        [DataTestMethod]
+        [DataRow("jquery@3.3.1", "jquery", "3.3.1")]
+        [DataRow("@angular/cli@1.0.0", "@angular/cli","1.0.0")]
+        [DataRow("My@Random@Library@1.0.0-preview3-final", "My@Random@Library", "1.0.0-preview3-final")]
+        [DataRow("@MyLibraryWithoutVersion", "@MyLibraryWithoutVersion", "")]
+        [DataRow("Library@Version", "Library", "Version")]
+        [DataRow(null, "", "")]
+        [DataRow("", "", "")]
+        public void GetLibraryNameAndVersion(string libraryId, string expectedName, string expectedVersion)
```

Comment:
What about `[DataRow("PartialId@", "PartialId", "")]` or similar to cover the case when it's only partially specified?  How should this handle the `@`?

Reply:
Right now if libraryId is `Partial@` it will return name as `Partial@` and Version as an empty string. 
That is in order to make sure we persist the `@` when writing libraryId back to the libman.json file.
Always adding an `@` while writing to file causes issues when the libraryId was just `Partial` to begin with.

Reply:
I see, that makes sense.  Can we have a test that verifies that behavior?  This might not be the right place to enforce it, but it sounds like it's a subtle scenario we wouldn't want to regress.